integration-tests/paho-mqtt5/src/test/java/org/apache/camel/quarkus/component/paho/mqtt5/it/PahoMqtt5TestResource.java

```diff
@@ -32,25 +36,76 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(PahoMqtt5TestResource.class);
     private static final String IMAGE = "eclipse-mosquitto:1.6.12";
     private static final int TCP_PORT = 1883;
+    private static final int SSL_PORT = 8883;
+    private static final int WS_PORT = 9001;
+    private static final String MQTT_USERNAME = "quarkus";
+    private static final String MQTT_PASSWORD = "quarkus";
 
     private GenericContainer<?> container;
+    private boolean useFixedPort = false;
+    private boolean startContainer = true;
+
+    @Override
+    public void init(Map<String, String> initArgs) {
+        initArgs.forEach((name, value) -> {
+            if (name.equals("useFixedPort")) {
+                useFixedPort = Boolean.parseBoolean(value);
+            } else if (name.equals("startContainer")) {
+                startContainer = Boolean.parseBoolean(value);
+            }
+        });
+    }
+
+    @Override
+    public void inject(TestInjector testInjector) {
+        testInjector.injectIntoFields(container,
+                new TestInjector.AnnotatedAndMatchesType(InjectPahoContainer.class, GenericContainer.class));
+    }
 
     @Override
     public Map<String, String> start() {
         LOGGER.info(TestcontainersConfiguration.getInstance().toString());
 
         try {
-            container = new GenericContainer<>(IMAGE)
-                    .withExposedPorts(TCP_PORT)
-                    .withLogConsumer(new Slf4jLogConsumer(LOGGER))
+            Map<String, String> result = new HashMap<>();
+
+            if (useFixedPort) {
```

Comment:
Could you please describe why we need to have use `useFixedPort` ? When should we use it ? And when not to use it please ?

Reply:
Ok, So it's only for manual testing. And yes, I know this re-connection scenario is hard to setup :) Many thanks.